Add custom metrics for FederatedHPA (#378)
karmada-io/website: i18n/zh/docusaurus-plugin-content-docs/current/tutorials/autoscaling-with-custom-metrics.md

---
title: FederatedHPA scales with custom metrics
---
In Karmada, a FederatedHPA scales up/down the workload's replicas across multiple clusters, with the aim of automatically scaling the workload to match the demand.

FederatedHPA not only supports resource metrics such as CPU and memory, but also supports custom metrics
which may expand the use cases of FederatedHPA.

This document walks you through an example of enabling FederatedHPA to automatically manage scale for a cross-cluster app with custom metrics.

The walkthrough example will do as follows:  

> ![federatedhpa-custom-metrics-demo](../resources/tutorials/custom_metrics.png)

Reviewer: ```suggestion

![federatedhpa-custom-metrics-demo](../resources/tutorials/custom_metrics.png)

```
Adding a blank line around the image can avoid cross-line on a big screen.

* One sample-deployment's pod exists in `member1` cluster.
* The service is deployed in `member1` and `member2` cluster.

> * Request the multi-cluster service and trigger the pod's custom metrics(http_requests_total) increases.

Reviewer: ```suggestion
* Request the multi-cluster service and trigger an increase in the pod's custom metrics (http_requests_total).
```

* The replicas will be scaled up in `member1` and `member2` cluster.

## Prerequisites

### Karmada has been installed

> We can install Karmada by referring to [Quick Start](https://github.com/karmada-io/karmada#quick-start), or directly run `hack/local-up-karmada.sh` script which is also used to run our E2E cases.

Reviewer: ```suggestion
You can install Karmada by referring to [Quick Start](https://github.com/karmada-io/karmada#quick-start),
or directly run `hack/local-up-karmada.sh` script which is also used to run your E2E cases.
```
